blazar/plugins/oshosts/host_plugin.py

```diff
@@ -1005,21 +1006,36 @@ def _poll_resource_failures(self):
         :return: a list of failed hosts, a list of recovered hosts.
         """
         hosts = db_api.host_get_all_by_filters({})
-        reservable_hosts = [h for h in hosts if h['reservable'] is True]
-        unreservable_hosts = [h for h in hosts if h['reservable'] is False]
+        reservable_ironic_hosts = [h for h in hosts if h['reservable'] is True and h['hypervisor_type'] == 'ironic']
+        unreservable_irinic_hosts = [h for h in hosts if h['reservable'] is False and h['hypervisor_type'] == 'ironic']
+        reservable_nova_hosts = [h for h in hosts if h['reservable'] is True and h['hypervisor_type'] != 'ironic']
+        unreservable_nova_hosts = [h for h in hosts if h['reservable'] is False and h['hypervisor_type'] != 'ironic']
+
+        failed_hosts = []
+        recovered_hosts = []
 
         try:
+            ironic_client = ironic.BlazarIronicClient()
+            bms = ironic_client.ironic.node.list()
+
+            failed_bm_ids = [bm.uuid for bm in bms
+                             if bm.maintenance or bm.power_state in ['error', None] or bm.provision_state not in ['active', 'available', None]]
```

Comment:
What does it mean for power_state or provision_state to be `None`?

Reply:
I'm not sure, but I'm following https://github.com/ChameleonCloud/nova/blob/chameleoncloud/train/nova/virt/ironic/driver.py#L259-L275